### PR TITLE
Added confirmation email field on checkout page.

### PR DESCRIPTION
**Pull request by LucaPipolo**
### Ticket
- [Prevent Users from entering wrong eMail adresses](https://app.asana.com/0/106767930274882/1173393553044193)

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -48,6 +48,7 @@ public static function preInit() {
     add_filter('woocommerce_variable_sale_price_html', __NAMESPACE__ . '\WooCommerce::woocommerce_variable_sale_price_html', 10, 2);
     add_filter('woocommerce_variable_price_html', __NAMESPACE__ . '\WooCommerce::woocommerce_variable_sale_price_html', 10, 2);
 
+
```
Comment (sun) [inline src/Plugin.php:51]:
```suggestion
```

File: src/WooCommerceCheckout.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   */
+  public static function woocommerceCheckoutSettings(array $settings): array {
+    $settings[] = [
+      'type' => 'title',
+      'name' => __('Checkout settings', Plugin::L10N),
+    ];
+    $settings[] = [
+      'type' => 'checkbox',
+      'id' => '_' . Plugin::L10N . '_checkout_email_confirmation_field',
+      'name' => __('Enable e-mail confirmation field', Plugin::L10N),
```
Comment (sun) [inline src/WooCommerceCheckout.php:40]: A confirmation is something else – usually means an actual verification email containing a link to click.
```suggestion
      'name' => __('Require users to enter billing email address twice', Plugin::L10N),
```

File: languages/shop-standards-de_DE.po [hunk 3]
```diff
@@ -237,11 +237,29 @@ msgstr ""
...
+msgid "Checkout settings"
+msgstr "Einstellungen Bestellprozess"
+
+#: src/WooCommerceCheckout.php:40
+msgid "Enable e-mail confirmation field"
+msgstr "E-Mail-Bestätigung aktivieren"
+
+#: src/WooCommerceCheckout.php:63
+msgid "Confirm e-mail address"
+msgstr "E-Mail-Adresse bestätigen"
```
Comment (sun) [inline languages/shop-standards-de_DE.po:255]:
```suggestion
msgid "Repeat email address"
msgstr "E-Mail-Adresse wiederholen"
```

File: src/WooCommerceCheckout.php (hunk 2 above)
Comment (LucaPipolo) [reply src/WooCommerceCheckout.php:40]: @stephansperling — should I change it?

File: languages/shop-standards-de_DE.po (hunk 3 above)
Comment (LucaPipolo) [reply languages/shop-standards-de_DE.po:255]: @stephansperling — should I change it?

File: src/WooCommerceCheckout.php (hunk 2 above)
Comment (stephansperling) [reply src/WooCommerceCheckout.php:40]: Please! Fine for me!

File: languages/shop-standards-de_DE.po (hunk 3 above)
Comment (stephansperling) [reply languages/shop-standards-de_DE.po:255]: @LucaPipolo please! Fine for me! :-)

File: languages/shop-standards.pot [hunk 4]
```diff
@@ -216,11 +216,27 @@ msgstr ""
...
 #, php-format
 msgid "from %1$s"
 msgstr ""
 
+#: src/WooCommerceCheckout.php:35
+msgid "Checkout settings"
+msgstr ""
+
+#: src/WooCommerceCheckout.php:40
+msgid "Enable e-mail confirmation field"
```
Comment (sun) [inline languages/shop-standards.pot:229]:
```suggestion
msgid "Require users to enter billing email address twice"
```

File: languages/shop-standards-de_DE.po [hunk 5]
```diff
@@ -237,11 +237,29 @@ msgstr ""
...
 msgid "from %1$s"
 msgstr "ab %1$s"
 
+#: src/WooCommerceCheckout.php:35
+msgid "Checkout settings"
+msgstr "Einstellungen Bestellprozess"
+
+#: src/WooCommerceCheckout.php:40
+msgid "Enable e-mail confirmation field"
+msgstr "E-Mail-Bestätigung aktivieren"
```
Comment (sun) [inline languages/shop-standards-de_DE.po:251]:
```suggestion
msgid "Require users to enter billing email address twice"
msgstr "Nutzer zur erneuten Eingabe der E-Mail-Adresse auffordern"
```

File: languages/shop-standards.pot [hunk 6]
```diff
@@ -216,11 +216,27 @@ msgstr ""
...
+msgstr ""
+
+#: src/WooCommerceCheckout.php:63
+msgid "Confirm e-mail address"
```
Comment (sun) [inline languages/shop-standards.pot:233]:
```suggestion
msgid "Repeat email address"
```

File: src/WooCommerceCheckout.php [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+   *
+   * @return array
+   *   The modified checkout fields.
+   *
+   * @implements woocommerce_checkout_fields
+   */
+  public static function addConfirmationEmailCheckoutField(array $fields): array {
+    $fields['billing']['billing_email']['class'] = ['form-row-first'];
+    $fields['billing']['billing_email_confirmation'] = [
+      'label' => __('Confirm e-mail address', Plugin::L10N),
```
Comment (sun) [inline src/WooCommerceCheckout.php:63]:
```suggestion
      'label' => __('Repeat email address', Plugin::L10N),
```

File: languages/shop-standards-de_DE.po [hunk 8]
```diff
@@ -289,3 +307,6 @@ msgstr "Herr"
 #: src/WooCommerceSalutation.php:67
 msgid "Company"
 msgstr "Firma"
+
+#~ msgid "Enable e-mail confirmation field"
+#~ msgstr "E-Mail-Bestätigung aktivieren"
```
Comment (juanlopez4691) [inline languages/shop-standards-de_DE.po:312]: This lines should be removed, right?

**Pull request merged**